{
  try {
    handler.handleException(exp);
  }
  finally {
    setException(exp);
  }
}
